iis-wf/iis-wf-ingest-pmc/src/test/java/eu/dnetlib/iis/wf/ingest/pmc/metadata/MetadataImporterMain.java

```diff
@@ -0,0 +1,84 @@
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+
+import eu.dnetlib.iis.ingest.pmc.metadata.schemas.ExtractedDocumentMetadata;
+
+/**
+ * This class is responsible for performing metadata extraction out of the JATS XML file which location is provided as input parameter.
+ * @author mhorst
+ */
+public class MetadataImporterMain {
+    
+    private static final String encoding = "utf-8"; 
+    
+    public static void main(String[] args) throws IOException, ParserConfigurationException, SAXException {
+        if (args.length >= 1) {
+            if (StringUtils.isBlank(args[0])) {
+                throw new RuntimeException("No valid location provided!");
+            }
+            
+            ExtractedDocumentMetadata.Builder builder  = ExtractedDocumentMetadata.newBuilder();
+            builder.setId("");
+            builder.setText("");
+            extractMetadata(getFileContent(args[0]), builder);
+            
+            DatumWriter<ExtractedDocumentMetadata> writer = new SpecificDatumWriter<>(ExtractedDocumentMetadata.class);
+            ByteArrayOutputStream stream = new ByteArrayOutputStream();
+            Encoder jsonEncoder = EncoderFactory.get().jsonEncoder(ExtractedDocumentMetadata.getClassSchema(), stream);
+            writer.write(builder.build(), jsonEncoder);
+            jsonEncoder.flush();
+            System.out.println(new String(stream.toByteArray(), encoding));
+            
+        } else {
+            throw new RuntimeException("input file path is missing");
+        }
+    }
+    
+    /**
+     * Extracts metadata from given xml input by supplementing metada in output builder.
+     * 
+     * @param xmlInput JATS XML file content
+     * @param output extracted metadata
+     * @throws ParserConfigurationException
+     * @throws SAXException
+     * @throws IOException
+     */
+    protected static void extractMetadata(String xmlInput, ExtractedDocumentMetadata.Builder output)
+            throws ParserConfigurationException, SAXException, IOException {
+        SAXParserFactory saxFactory = SAXParserFactory.newInstance();
+        saxFactory.setValidating(false);
+        SAXParser saxParser = saxFactory.newSAXParser();
+        XMLReader reader = saxParser.getXMLReader();
+        reader.setFeature("http://xml.org/sax/features/validation", false);
+        reader.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+        reader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+        JatsXmlHandler pmcXmlHandler = new JatsXmlHandler(output);
+        saxParser.parse(new InputSource(new StringReader(xmlInput)), pmcXmlHandler);
+    }
+    
+    /**
+     * Reads file content from the location provided as input parameter.
+     */
+    private static String getFileContent(String location) throws IOException {
+        return new String(Files.readAllBytes(Paths.get(location)), Charset.forName(encoding));
+    }
```

Comment:
Not sure if going `byte[]` -> `String` -> `StringReader` is better than `byte[]` -> `ByteArrayInputStream`. As long as the file encoding is actually UTF-8 both should work.

Reply:
I think both methods were copied from different places what introduced this conversion overhead. I've simplified both methods a little bit in order to stick to simple byte[] representation.